### ISSUE TITLE
bug: `melta.no` does not support HTTPS

### ISSUE THREAD

**Issue by WillDaSilva** [meltano/meltano#6804]
### Meltano Version

N/A

### Python Version

NA

### Bug scope

Other

### Operating System

N/A

### Description

[http://melta.no works](http://melta.no).

[https://melta.no does not work](https://melta.no):

![image](https://user-images.githubusercontent.com/11428666/192583754-f7201304-acae-4fd9-94ad-5115ae09edf0.png)

From @tayloramurphy in https://github.com/meltano/meltano/pull/6748#issuecomment-1248406910:

> I've emailed the domain provider to see about getting an SSL Certificate. http://melta.no/ does work and forwards appropriately from what I see. FYI @therebbie I put an inquiry through the Marcaria contact form to go to the hello meltano address.

**Comment by therebbie** [comment]
@WillDaSilva I thought we ran into this a couple of weeks ago and via a different issue (not the one listed above).

@tayloramurphy the hello address forwards (I believe) to Zendesk for such things. In this case, Marcaria responded thusly:
> To add HTTPS to a domain, you need hosting. The SSL issuer needs to verify that the domain's host is valid. Maybe check with your current hosting provider about the option to host [melta.no](http://melta.no/), but with just URL forward is not possible even with frame-forward; it will show a not secure connection for [melta.no](http://melta.no/) and [meltano.com](http://meltano.com/)'s website.

**Comment by visch** [comment]
fwiw @therebbie  Google Domains auto sets up an SSL cert for you an will forward your subdomain (or domains) to whatever pages you want. Maybe others do this too, but might be easier then setting up something on aws and managing it 🤷

**Comment by WillDaSilva** [comment]
> To add HTTPS to a domain, you need hosting. The SSL issuer needs to verify that the domain's host is valid. Maybe check with your current hosting provider about the option to host [melta.no](http://melta.no/), but with just URL forward is not possible even with frame-forward; it will show a not secure connection for [melta.no](http://melta.no/) and [meltano.com](http://meltano.com/)'s website.

@therebbie @tayloramurphy So if I understand the situation correctly, we are currently using our domain registrar Marcaria to provide URL forwarding instead of managing the DNS directly with our regular DNS provider?

Seems like we should switch away from this URL forwarding service, and onto our regular DNS server. Then since CNAME is not an option because we are pointing it at a specific path under `docs.meltano.com`, we should set up a 301 redirect with our webserver (whatever that may be) with TLS support. We'll either need to get new TLS certs for `melta.no`, or get a multi-domain cert that supports both `docs.meltano.com` and `melta.no`.

**Comment by therebbie** [comment]
I believe you are correct above, @WillDaSilva ... FWIW, I don't manage any of the actual IT infrastructure but I am always a fan of keeping DNS and registrar registrations with different providers.

**Comment by pandemicsyn** [comment]
> Seems like we should switch away from this URL forwarding service, and onto our regular DNS server. Then since CNAME is not an option because we are pointing it at a specific path under docs.meltano.com, we should set up a 301 redirect with our webserver (whatever that may be) with TLS support. We'll either need to get new TLS certs for melta.no, or get a multi-domain cert that supports both docs.meltano.com and melta.no.

docs.meltano.com and friends already use Netlify , and I think' we've already got one or two domain aliases there.  So we can just mirror what we've done in the past and configure melta.no in Route53 and point it to Netlify. Netlify will take care of the SSL coverage and allow us to configure the redirect:

- https://docs.netlify.com/domains-https/https-ssl/#domain-aliases
- https://docs.netlify.com/domains-https/custom-domains/multiple-domains/#domain-redirects

**Comment by tayloramurphy** [comment]
@WillDaSilva @pandemicsyn so it sounds like we have a resolution here with Route53? Can this get a weight and we can get it prioritized? It's not clear what we'd have to do on the Marcaria side, but it looks like set up nameservers and dns records as needed. Happy to do a quick pair on this so we can get it across the finish line.

**Comment by WillDaSilva** [comment]
I can't find a way to lower the TTL with Marcaria. It's not essential - the current TTL is 21600 seconds (6 hours), so if something goes wrong during the transfer to Route53 `melta.no` may go down for up to 6 hours. Because of that, I'll do this during the upcoming weekend. That way there'll be minimal disruption is something goes wrong.

**Comment by WillDaSilva** [comment]
Current DNS records:

![image](https://user-images.githubusercontent.com/11428666/193919701-ba55debb-0c2b-4478-9cbb-0755a35f497f.png)

**Comment by WillDaSilva** [comment]
More useful links:
- https://docs.netlify.com/routing/redirects/redirect-options/#domain-level-redirects
- https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/migrate-dns-domain-in-use.html

**Comment by WillDaSilva** [comment]
@pandemicsyn @tayloramurphy I'll note every step I take here.

I created a new hosted zone for `melta.no` in Route53. It comes with default NS and SOA records.

Route53 does not support ALIAS/ANAME records, or CNAME flattening, so we cannot place a CNAME record at the root domain `melta.no` that points to `meltano.netlify.app` as we have done with `docs.meltano.com` and `discovery.meltano.com`.

I tried creating a new subdomain `docs.melta.no` which has a `CNAME` record pointing at `meltano.netlify.app`, and then made `melta.no` an alias for `docs.melta.no`, but Route53 only allows record aliases between records of the same type. 

I set an A record that points `melta.no` at Netlify's load balancer at `75.2.60.5`. Unfortunately Netlify does not provide an IPv6 load balancer: https://answers.netlify.com/t/aaaa-ipv6-record-for-apex-domains/11872/9

The nameservers used by Marcaria have been updated to the ones provided by Route53 (from the default NS record):

```
ns-1581.awsdns-05.co.uk (205.251.198.45)
ns-141.awsdns-17.com    (205.251.192.141)
ns-1443.awsdns-52.org   (205.251.197.163)
ns-552.awsdns-05.net    (205.251.194.40)
```

I added a domain alias for `melta.no` to our Netlify account. I will wait until `http://melta.no` is using Route53 (hopefully within 6 hours) to update the TLS certificate managed by Netlify to include `melta.no` in addition to `docs.meltano.com` and `discovery.meltano.com`.

The PR https://github.com/meltano/meltano/pull/6865 has been opened to configure the Netlify redirects from `melta.no` to `docs.meltano.com/guide/troubleshooting`. If this PR is not merged in before the DNS updates take effect, we should find `melta.no` routing to `docs.meltano.com`.
